etl/steps/data/garden/faostat/2023-02-22/detected_anomalies.py

```diff
@@ -0,0 +1,358 @@
+"""Detected anomalies.
+
+This module contains a class for each data anomaly detected. The class has the following methods and properties:
+* description: A human-readable text that describes the anomaly.
+  NOTE: The description will be added to the dataset metadata description, and hence will be shown in grapher.
+* checks: A method that ensures the anomaly exists in the data.
+  This is useful to detect if an anomaly has been corrected after a data update.
+* fix: A method that removes the anomaly.
+* inspect: An optional method that plots (in the browser) a visualization that shows the anomaly.
+  It can be used before and after removing the anomalies.
+* handle_anomalies: A helper method that uses all the previous methods in the usual order.
+
+If after an update the anomaly is no longer in the original data, remove the corresponding class.
+
+"""
+import abc
+import os
+from typing import Tuple
+
+import pandas as pd
+import plotly.express as px
+from structlog import get_logger
+
+log = get_logger()
+
+# If environment variable INSPECT_ANOMALIES is set to True, run the step in interactive mode.
+INSPECT_ANOMALIES = bool(os.getenv("INSPECT_ANOMALIES", False))
+
+
+class DataAnomaly(abc.ABC):
+    """Abstract class for data anomaly."""
+
+    def __init__(self) -> None:
+        pass
+
+    @property
+    @abc.abstractmethod
+    def description(self) -> str:
+        # Description of the anomaly, which will be added to the dataset description.
+        pass
+
+    @abc.abstractmethod
+    def check(self, df: pd.DataFrame) -> None:
+        pass
+
+    @abc.abstractmethod
+    def fix(self, df: pd.DataFrame) -> pd.DataFrame:
+        pass
```

Comment:
Instead of using `pass`, I'd raise a `NotImplementedError` exception to ensure this method is never called from the abstract class.

From the python [docs](https://docs.python.org/3/library/exceptions.html#NotImplementedError):

> In user defined base classes, abstract methods should raise this exception when they require derived classes to override the method, or while the class is being developed to indicate that the real implementation still needs to be added.